sklearn/utils/_metadata_requests.py

```diff
@@ -0,0 +1,1201 @@
+        corresponding method as extra props.
+
+        Parameters
+        ----------
+        method : str
+            The name of the method for which the parameters are requested and
+            routed.
+
+        params : dict
+            A dictionary of provided metadata.
+
+        Returns
+        -------
+        params : Bunch
+            A :class:`~utils.Bunch` of {prop: value} which can be given to the
+            corresponding method.
+        """
+        res = Bunch()
+        if self._self:
+            res.update(self._self._get_params(params=params, method=method))
+        param_names = self._get_param_names(
+            method=method, original_names=False, ignore_self=True
+        )
+        res.update({key: value for key, value in params.items() if key in param_names})
+        return res
+
+    def get_params(self, *, method, params):
+        """Return the input parameters requested by a child objects.
+
+        The output of this method is a bunch, which includes the inputs for all
+        methods of each child object that are used in the router's `"method"`.
+
+        If the router is also a consumer, it also checks for warnings of
+        `self`'s/consumer's requested metadata.
+
+        Parameters
+        ----------
+        method : str
+            The name of the method for which the parameters are requested and
+            routed.
+
+        params : dict
+            A dictionary of provided metadata.
+
+        Returns
+        -------
+        params : Bunch
+            A :class:`~utils.Bunch` of the form
+            ``{"object_name": {"method_name": {prop: value}}}`` which can be
+            used to pass the required metadata to corresponding methods or
+            corresponding child objects.
+        """
+        if self._self:
+            self._self._check_warnings(params=params, method=method)
+
+        res = Bunch()
+        for name, route_mapping in self._route_mappings.items():
+            router, mapping = route_mapping.routing, route_mapping.mapping
+
+            res[name] = Bunch()
+            for orig_method, used_in in mapping:
+                if used_in == method:
+                    if isinstance(router, MetadataRequest):
+                        res[name][orig_method] = router._get_params(
+                            params=params, method=orig_method
+                        )
+                    else:
+                        res[name][orig_method] = router._get_squashed_params(
+                            params=params, method=orig_method
+                        )
+        return res
+
+    def validate_metadata(self, *, method, params):
+        """Validate given metadata for a method.
+
+        This raises a ``ValueError`` if some of the passed metadata are not
+        understood by child objects.
+
+        Parameters
+        ----------
+        method : str
+            The name of the method for which the parameters are requested and
```

Comment:
as above, not clearer if this is the source or destination method name.

Reply:
Tried to clarify, but also note that this is not called by users if they use the decorator.